test/unit/constants.test.ts

```diff
@@ -36,20 +37,95 @@ describe("constants", () => {
     })
   })
   describe("version", () => {
-    it("should return the package.json version", () => {
-      // Source: https://gist.github.com/jhorsman/62eeea161a13b80e39f5249281e17c39#gistcomment-2896416
-      const validSemVar = new RegExp("^(0|[1-9]d*).(0|[1-9]d*).(0|[1-9]d*)")
-      const isValidSemVar = validSemVar.test(version)
-      expect(version).not.toBe(null)
-      expect(isValidSemVar).toBe(true)
+    describe("with package.json.version defined", () => {
+      let mockPackageJson = {
+        name: "mock-code-server",
+        version: "1.0.0",
+      }
+      let version = ""
+
+      beforeEach(() => {
```

Comment:
NBD but all these `beforeEach`  and `afterEach` blocks are either identical or nearly identical so thoughts on combining them and moving into a common describe block above? Would make adding future tests a bit smoother I think.

To combine the nearly identical ones we could do 
```
constants = require("../../src/node/constants")
```

And then the test uses whichever property it's trying to test.

Reply:
It did feel a bit repetitive 😅 I didn't bother thinking of a better solution while writing it but I like the idea of a common describe block above! Let me give that a try and see how it works 🚀

Reply:
Fixed!